### PR TITLE
Sync engine migration

### PR DESCRIPTION
**Pull request by gizemb**
### Issues

Having [wire-android ](https://github.com/wireapp/wire-android/)(UI) and [wire-android-sync-engine](https://github.com/wireapp/wire-android-sync-engine) (SE) as separate projects causes problems when a feature requires a change in both projects. In such a case, we first need to make changes in SE, then publish the SE project as a jar to local maven repo, then reference this published jar as a dependency to be able to work on it in the UI project. 

### Solutions

To overcome this impractical cycle, we decided to merge SE and UI projects under same project, by adding SE code as a module inside UI project.

### Testing

Full regression test is needed to guarantee that no functionality is changed.

## Notes

After this merge, wire-android-sync-engine project won't be taking any updates, and all necessary code will be pushed to this repository.

### CODE REVIEW

File: wire-android-sync-engine/build.sbt [hunk 1]
```diff
@@ -0,0 +1,191 @@
+//import java.io.File
```
Comment (johnxnguyen) [inline wire-android-sync-engine/build.sbt]: Should we delete this?

File: wire-android-sync-engine/zmessaging/src/main/res/values/random_words.xml [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+-->
+<resources>
+    <string-array name="random_adjectives">
+        <item>acid</item>
```
Comment (johnxnguyen) [inline wire-android-sync-engine/zmessaging/src/main/res/values/random_words.xml]: Ahaha, what's this?

Comment (marcoconti83) [reply wire-android-sync-engine/zmessaging/src/main/res/values/random_words.xml]: It's needed for random user handles creation, like `RedPanda43`

File: wire-android-sync-engine/build.sbt (hunk 1 above)
Comment (gizemb) [reply wire-android-sync-engine/build.sbt]: Yes, I forgot! Done on the next commit

**Comment by marcoconti83** [comment]
build again

**Comment by gizemb** [comment]
build again

**Comment by gizemb** [comment]
build again

**Comment by gizemb** [comment]
build again

**Comment by gizemb** [comment]
build again

**Comment by gizemb** [comment]
build again

**Comment by gizemb** [comment]
build again

**Pull request merged**